### PR TITLE
DataType Changes, validation output transpose

### PR DESCRIPTION
**Pull request by najiyacl**
## Description

- Convert Pandas dataframe  when its passed with with_schema_details….. and use Spark Type to Feature Type mapping
- Validate input feature details, subset
- Change FeatureType.UNKNOWN to ….FeatureType.COMPLEX for not mapped spark types
- validation output transpose

## Unit tests
Screenshot 2023-07-05 at 11 19 21 AM

## Integration tests
(https://teamcity.oci.oraclecorp.com/buildConfiguration/Odsc_FeatureStore_FeatureStoreAdsIntegrationTests_FeatureStoreAdsIntegrationTestsPython38/40386914?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true)

### CODE REVIEW

File: ads/feature_store/common/enums.py [hunk 1]
```diff
@@ -295,6 +295,7 @@ class FeatureType(Enum):
     STRING_BINARY_MAP = "STRING_BINARY_MAP"
     STRING_BOOLEAN_MAP = "STRING_BOOLEAN_MAP"
     UNKNOWN = "UNKNOWN"
+    COMPLEX = "COMPLEX"
```
Comment (KshitizLohia) [inline ads/feature_store/common/enums.py:298]: let's also add 1.1 in release_notes.rst and update the SERVICE_VERSION after merging the backend PR and add release notes to update the datatype to COMPLEX.

File: tests/integration/feature_store/test_datatype_pandas_mixed.py [hunk 2]
```diff
@@ -5,31 +5,49 @@
...
-                .with_name(self.get_name("feature_group_pandas_mixed"))
-                .with_entity_id(entity_id)
-                .with_feature_store_id(feature_store_id)
-                .with_primary_keys([])
-                .with_schema_details_from_dataframe(flights_df_mixed)
-                .with_statistics_config(False)
-            )
+        return feature_group_pandas_mixed
+
+    def define_feature_group_resource_with_pandas_mixed_infer_schema_nan(
```
Comment (KshitizLohia) [inline tests/integration/feature_store/test_datatype_pandas_mixed.py:35]: is it possible to incorporate an integration test case that specifically addresses the issue of the malfunctioning MLM library?

Comment (najiyacl) [reply tests/integration/feature_store/test_datatype_pandas_mixed.py:35]: Not sure of the root cause here to add the test case.Just happened to find while testing the UGBU

File: ads/feature_store/common/utils/utility.py [hunk 3]
```diff
@@ -268,3 +260,45 @@ def largest_matching_subset_of_primary_keys(left_feature_group, right_feature_gr
...
+
+
+def validate_spark_dataframe_schema(raw_feature_details: List[dict], input_df: DataFrame):
+    feature_detail_map = {}
+    columns_to_remove = []
+    for feature_details in raw_feature_details:
+        feature_detail_map[feature_details.get("name")] = feature_details
+    for column in input_df.columns:
+        if column not in feature_detail_map.keys():
+            logger.warning("column" + column + "doesnt exist in the input feature details")
```
Comment (harsh97) [inline ads/feature_store/common/utils/utility.py:295]: nit: doesn't

File: ads/feature_store/common/utils/utility.py [hunk 4]
```diff
@@ -268,3 +260,45 @@ def largest_matching_subset_of_primary_keys(left_feature_group, right_feature_gr
...
+                .astype(pandas_type)
+                .where(pd.notnull(input_df[column]), None)
+            )
+        else:
+            logger.warning("column" + column + "doesnt exist in the input feature details")
+            columns_to_remove.append(column)
+    return input_df.drop(columns = columns_to_remove)
+
+
+def validate_spark_dataframe_schema(raw_feature_details: List[dict], input_df: DataFrame):
```
Comment (harsh97) [inline ads/feature_store/common/utils/utility.py:288]: should we call it validate given that it is changing the dataframe itself. Also, we can add the return type for the function as well here

File: ads/feature_store/docs/source/feature_group.rst [hunk 5]
```diff
@@ -175,10 +175,6 @@ Statistics Results
 ==================
 You can call the ``get_statistics()`` method of the FeatureGroup instance to fetch validation results for a specific ingestion job.
 
-.. note::
-
-  PyDeequ is a Python API for Deequ, a library built on top of Apache Spark for defining "unit tests for data", which measure data quality in large datasets.
```
Comment (harsh97) [inline ads/feature_store/docs/source/feature_group.rst:180]: :+1:

File: tests/integration/feature_store/test_dataset_validations.py [hunk 6]
```diff
@@ -55,7 +55,7 @@ def test_dataset_validation_operations(self):
         assert dataset.oci_dataset.id
 
         dataset.materialise()
-        df = dataset.get_validation_output().to_pandas()
+        df = dataset.get_validation_output().to_pandas().T
```
Comment (harsh97) [inline tests/integration/feature_store/test_dataset_validations.py:58]: did you mean to call transpose function here?

Comment (najiyacl) [reply tests/integration/feature_store/test_dataset_validations.py:58]: since it checks the column value,or we would need to change the logic in the test

**Review by harsh97 — APPROVED** [review]

**Review by KshitizLohia — APPROVED** [review]

**Pull request merged**